internal/controlplane/handlers_authz.go

```diff
@@ -54,27 +47,12 @@ func isSuperadmin(claims auth.UserPermissions) bool {
 	return claims.IsStaff
 }
 
-func containsSuperadminRole(openIdToken openid.Token) bool {
-	if realmAccess, ok := openIdToken.Get("realm_access"); ok {
-		if realms, ok := realmAccess.(map[string]interface{}); ok {
-			if roles, ok := realms["roles"]; ok {
-				if userRoles, ok := roles.([]interface{}); ok {
-					if slices.Contains(userRoles, "superadmin") {
-						return true
-					}
-				}
-			}
-		}
-	}
-	return false
-}
-
 // lookupUserPermissions returns the user permissions from the database for the given user
-func lookupUserPermissions(ctx context.Context, store db.Store, tok openid.Token) (auth.UserPermissions, error) {
+func lookupUserPermissions(ctx context.Context, store db.Store, subject string) (auth.UserPermissions, error) {
```

Comment:
If we have `GetUserSubjectFromContext`, do we need to pass in subject?

Reply:
We don't, I refactored